Plugins/Json/include/Acts/Plugins/Json/GeometryHierarchyMapJsonConverter.hpp

```diff
@@ -114,15 +142,19 @@ nlohmann::json GeometryHierarchyMapJsonConverter<value_t>::toJson(
   nlohmann::json entries = nlohmann::json::array();
   for (std::size_t i = 0; i < container.size(); ++i) {
     auto entry = encodeIdentifier(container.idAt(i));
-    entry["value"] = nlohmann::json(container.valueAt(i));
+    auto value_json = nlohmann::json(container.valueAt(i));
+    if constexpr (not std::is_same<decorator_t, void>::value) {
```

Comment:
Just an idea, but if you want to get rid of the ```if constexpr```, then it might be possible to add an overload to ```decorateJson``` for ```decorator_t = void``` that does nothing? Or would that break the ```missing_specialization``` pattern?

Reply:
I do not really have a strong opinion on this. 
In both cases the compiler is needed to optimise away the code which does nothing. I thought that in this way it is more explicit when decorations are applied and when not. A e.g. "do_nothing" decorator still could do something. So its implementation would have to be checked. On the other hand a "void" decorator can absolutely not do anything.

Reply:
Absolutely fine with me, I leave it to you